client/src/pages/Edit.js

```diff
@@ -13,18 +18,17 @@ function Edit() {
 
   useEffect(() => {
     const editFoodId = async () => {
-      const response = await fetch("http://localhost:5000/food");
+      const response = await fetch("http://localhost:5000/menu_items"); //change endpoint to hit 1 entry not all
```

Comment:
You probably need a singular `menu_item` endpoint that also supports a `GET` and requires an `id`.